js/internal/subscriptions.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+import {NativeEventSubscription} from 'react-native';
+import NativeInterface from './nativeInterface';
+import * as Types from './types';
+
+const DEVICE_CONNECTIVITY_EVENT = 'netInfo.networkStatusDidChange';
+
+const _subscriptions = new Set<Types.NetInfoChangeHandler>();
```

Comment:
I may have no idea what I'm talking about here, but is instance or static the right way to handle listeners for the module? Is it a singleton so instance works, or maybe static isn't right because it's possible to have multiple independent sets, or...I may have no idea what I'm talking about

Reply:
I *think* I know what you're asking here, but correct me if I'm wrong.

Because of the way that Javascript modules work, we can guarantee that there is only one `_subscriptions` `Set` which is created the first time the module is imported. We are creating this the first time and only managing that with functions in that module.

We could create a class and access it using a singleton pattern, but that wouldn't actually give us anything else except for wrapping it in a class and maybe making it slightly easier to test without needing to "reset" it each time.

For now, I think we can leave it as it is and we can look into it in the future. It's deliberately an `internal` detail of the library as it's something I'm not 100% happy with, but it works for now.

Reply:
Yeah - that was pretty much it - this specifically:

> Because of the way that Javascript modules work, we can guarantee that there is only one _subscriptions Set which is created the first time the module is imported

As a javascript novice (I'm a Java guys since :lizard:'s roamed) I am iffy in this area. Seems solid then